tests/filtered_relation/tests.py

```diff
@@ -707,6 +710,139 @@ def test_eq(self):
             FilteredRelation("book", condition=Q(book__title="b")), mock.ANY
         )
 
+    def test_conditional_expression(self):
+        self.assertSequenceEqual(
+            Author.objects.annotate(
+                the_book=FilteredRelation(
+                    "book",
+                    condition=Q(Value(False)),
+                ),
+            ).filter(
+                the_book__isnull=False,
+            ),
+            [],
+        )
+
+    def test_expression_outside_relation_name(self):
+        qs = Author.objects.annotate(
+            book_editor=FilteredRelation(
+                "book__editor",
+                condition=Q(
+                    Exact(F("book__author__name"), "Alice"),
+                    Value(True),
+                    book__title__startswith="Poem",
+                ),
+            ),
+        ).filter(book_editor__isnull=False)
+        self.assertEqual(qs.count(), 1)
+
+    def test_conditional_expression_with_case(self):
+        qs = Book.objects.annotate(
+            alice_author=FilteredRelation(
+                "author",
+                condition=Q(
+                    Case(
+                        When(
+                            author__name="Alice",
+                            then=True,
+                        ),
+                        default=False,
+                    )
+                ),
+            ),
+        ).filter(alice_author__isnull=False)
+        self.assertEqual(qs.count(), 2)
+
+    def test_conditional_expression_with_case_outside_relation_name(self):
+        qs = (
+            Author.objects.annotate(
+                book_editor=FilteredRelation(
+                    "book__editor",
+                    condition=Q(
+                        Case(
+                            When(
+                                book__author__name="Alice",
+                                then=True,
+                            ),
+                            default=False,
+                        )
+                    ),
+                ),
+            )
+            .filter(book_editor__isnull=True)
+            .distinct()
+        )
+        self.assertEqual(qs.count(), 1)
+
+    def test_conditional_expression_with_lookup(self):
+        qs = (
+            Author.objects.annotate(
+                poem_book=FilteredRelation(
+                    "book",
+                    condition=Q(IStartsWith(F("book__title"), "poem")),
+                ),
+            )
+            .filter(poem_book__isnull=False)
+            .distinct()
```

Comment:
`distinct()` should not be needed :thinking: As far as I'm aware,
```
condition=Q(IStartsWith(F("book__title"), "poem"))
``` 
should work exactly the same as
```
condition=Q(book__title__istartswith="poem")
```
and returns a singe author. It is also suspicious to use `distinct()` in other tests.